tests/template_tests/syntax_tests/i18n/test_blocktrans.py

```diff
@@ -0,0 +1,459 @@
+        self.assertEqual(output, 'α &amp; β')
+
+    @setup({'i18n18': '{% load i18n %}'
+                      '{% blocktrans with berta=anton|force_escape %}{{ berta }}{% endblocktrans %}'})
+    def test_i18n18(self):
+        output = self.engine.render_to_string('i18n18', {'anton': 'α & β'})
+        self.assertEqual(output, 'α &amp; β')
+
+    @setup({'i18n19': '{% load i18n %}{% blocktrans %}{{ andrew }}{% endblocktrans %}'})
+    def test_i18n19(self):
+        output = self.engine.render_to_string('i18n19', {'andrew': 'a & b'})
+        self.assertEqual(output, 'a &amp; b')
+
+    @setup({'i18n21': '{% load i18n %}{% blocktrans %}{{ andrew }}{% endblocktrans %}'})
+    def test_i18n21(self):
+        output = self.engine.render_to_string('i18n21', {'andrew': mark_safe('a & b')})
+        self.assertEqual(output, 'a & b')
+
+    @setup({'legacyi18n17': '{% load i18n %}'
+                            '{% blocktrans with anton|escape as berta %}{{ berta }}{% endblocktrans %}'})
+    def test_legacyi18n17(self):
+        output = self.engine.render_to_string('legacyi18n17', {'anton': 'α & β'})
+        self.assertEqual(output, 'α &amp; β')
+
+    @setup({'legacyi18n18': '{% load i18n %}'
+                            '{% blocktrans with anton|force_escape as berta %}'
+                            '{{ berta }}{% endblocktrans %}'})
+    def test_legacyi18n18(self):
+        output = self.engine.render_to_string('legacyi18n18', {'anton': 'α & β'})
+        self.assertEqual(output, 'α &amp; β')
+
+    @setup({'i18n26': '{% load i18n %}'
+                      '{% blocktrans with extra_field=myextra_field count counter=number %}'
+                      'singular {{ extra_field }}{% plural %}plural{% endblocktrans %}'})
+    def test_i18n26(self):
+        """
+        translation of plural form with extra field in singular form (#13568)
+        """
+        output = self.engine.render_to_string('i18n26', {'myextra_field': 'test', 'number': 1})
+        self.assertEqual(output, 'singular test')
+
+    @setup({'legacyi18n26': '{% load i18n %}'
+                            '{% blocktrans with myextra_field as extra_field count number as counter %}'
+                            'singular {{ extra_field }}{% plural %}plural{% endblocktrans %}'})
+    def test_legacyi18n26(self):
+        output = self.engine.render_to_string('legacyi18n26', {'myextra_field': 'test', 'number': 1})
+        self.assertEqual(output, 'singular test')
+
+    @setup({'i18n27': '{% load i18n %}{% blocktrans count counter=number %}'
+                      '{{ counter }} result{% plural %}{{ counter }} results'
+                      '{% endblocktrans %}'})
+    def test_i18n27(self):
+        """translation of singular form in Russian (#14126)"""
+        with translation.override('ru'):
+            output = self.engine.render_to_string('i18n27', {'number': 1})
+        self.assertEqual(
+            output, '1 \u0440\u0435\u0437\u0443\u043b\u044c\u0442\u0430\u0442')
```

Comment:
Wrapping wasn't reverted here (I didn't check the patch exhaustively perhaps you can give it another look).